src/services/deployments.service.ts

```diff
@@ -43,25 +43,30 @@ export class DeploymentsService {
 	}
 
 	static async listByProjectId(projectId: string): Promise<ServiceResponse<Deployment[]>> {
-		const { data: environments, error: environmentsError } = await EnvironmentsService.listByProjectId(projectId);
+		try {
+			const { data: environments, error: environmentsError } = await EnvironmentsService.listByProjectId(projectId);
 
-		if (environmentsError) {
-			LoggerService.error(namespaces.deploymentsService, (environmentsError as Error).message);
+			if (environmentsError) {
+				LoggerService.error(namespaces.deploymentsService, (environmentsError as Error).message);
 
-			return { data: undefined, error: environmentsError };
-		}
+				return { data: undefined, error: environmentsError };
+			}
 
-		const environmentIds = getIds(environments!, "envId");
-		const { data: projectDeployments, error: deploymentsError } = await this.listByEnvironmentIds(environmentIds);
+			const environmentIds = getIds(environments!, "envId");
+			const { data: projectDeployments, error: deploymentsError } = await this.listByEnvironmentIds(environmentIds);
 
-		if (deploymentsError) {
-			LoggerService.error(namespaces.deploymentsService, (deploymentsError as Error).message);
+			if (deploymentsError) {
```

Comment:
deployment error? are you trying to deploy something?

Reply:
refactored